kube-controllers/pkg/controllers/loadbalancer/loadbalancer_controller.go

```diff
@@ -671,13 +671,25 @@ func (c *loadBalancerController) assignIP(svc *v1.Service) ([]string, error) {
 		return nil, nil
 	}
 
+	// Get the namespace object for namespaceSelector support
+	var namespaceObj *v1.Namespace
+	if svc.Namespace != "" {
+		ns, err := c.clientSet.CoreV1().Namespaces().Get(context.Background(), svc.Namespace, metav1.GetOptions{})
+		if err != nil {
+			log.WithError(err).WithField("namespace", svc.Namespace).Warn("Failed to get namespace for LoadBalancer IP assignment, proceeding without namespace selector support")
```

Comment:
I think we'll want to just return an error here - if the Namespace isn't accessible, we shouldn't try to assign an IP.